.github/workflows/build_and_test.yml

```diff
@@ -130,46 +130,23 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    
+
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
         go-version-file: 'go.mod'
         cache: true
         cache-dependency-path: go.sum
-    
-    - name: Cache apt packages
-      uses: actions/cache@v4
-      with:
-        path: /var/cache/apt/archives
-        key: ${{ runner.os }}-apt-unit-tests-${{ hashFiles('.github/workflows/build_and_test.yml') }}
-        restore-keys: |
-          ${{ runner.os }}-apt-unit-tests-
-    
-    - name: Install build dependencies
-      run: |
-        sudo apt-get update -qq
-        sudo apt-get install -y -qq clang llvm libelf-dev libbpf-dev
-    
+
     - name: Run unit tests
       run: |
-        echo "Running unit tests with coverage (excluding integration tests)..."
-        # Run tests with coverage and save results (generate is a dependency of test-unit)
+        echo "Running unit tests (excluding integration tests)..."
         make test-unit 2>&1 | tee unit-test-results.txt
```

Comment:
This appears to be swallowing the exit code from `make test-unit` which means the job always passes even if the tests fail